text/0000-dynamic-attributes.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Passing Dynamic Attributes/Properties
+status: DRAFTED
+created_at: 2021-07-05
+updated_at: YYYY-MM-DD
+champion: Mohammed Abdul Sattar (abdulsattar)
+pr: https://github.com/salesforce/lwc-rfcs/pull/52
+---
+
+# Passing Dynamic Attributes/Properties
+
+## Summary
+This proposal adds mechanism to pass a dynamic attributes or properties bag to elements in
+an LWC template.
+
+## Example
+```js
+// x/myComponent.js
+export default class MyComponent extends LightningElement {
+    childProps = { 
+        name: 'My Child',
+        style: 'width: 50px',
+        class: 'my-class',
+        onclick: this.handleClick,
+    };
```

Comment:
I take it that for strings like `ariaDisabled`, this will automatically convert to `aria-disabled`? Will we support the kebab-cased versions as well, or only the camelCased versions?

There are also certain odd attributes like `aria-posinset` which [would look most natural in camelCase as `ariaPosInSet`](https://wicg.github.io/aom/spec/aria-reflection.html#idl_attr_exceptions), but I suppose we should prefer `ariaPosinset` to avoid having to keep a list of exceptions?

Reply:
Depending on what we decide here, this may also influence event handers, e.g. `onclick` versus `onClick`.

Reply:
We can just leave it as is: `childProps = { 'aria-posinset': '5' };` without any conversion.

Reply:
I see, so we wouldn't do any camelCase -> kabeb-case conversion? That's probably the simplest solution.

Reply:
I think this has changed since the last conversation, to use aria, you must use the prop name, otherwise it does nothing.